components/keys/src/lib.rs

```diff
@@ -210,6 +210,14 @@ pub fn data_key(key: &[u8]) -> Vec<u8> {
     v
 }
 
+pub fn data_key_with_buff<'a>(buf: &'a mut Vec<u8>, key: &[u8]) -> &'a [u8] {
+    let l = DATA_PREFIX_KEY.len() + key.len();
```

Comment:
This doesn't seem necessary.

Reply:
Yes.... I just want to avoid once memory allocation.

Reply:
I will remove it and add it in another PR.